src/vec_simd.rs

```diff
@@ -67,6 +81,100 @@ pub fn scaled_add(mut u: ArrayViewMut1<f32>, v: ArrayView1<f32>, a: f32) {
     )
 }
 
+#[cfg(all(target_arch = "aarch64", feature = "neon"))]
+pub mod neon {
+    use std::arch::aarch64::*;
+
+    use ndarray::{ArrayView1, ArrayViewMut1};
+
+    use crate::vec_simd::{dot_unvectorized, scale_unvectorized, scaled_add_unvectorized};
+
+    #[inline(always)]
+    #[allow(dead_code)]
+    pub(crate) unsafe fn neon_add(values: float32x4_t) -> f32 {
+        // vaddvq_f32 is missing in std::arch::aarch64?
+        let sums = vadd_f32(vget_high_f32(values), vget_low_f32(values));
+        vget_lane_f32::<0>(sums) + vget_lane_f32::<1>(sums)
+    }
+
+    #[target_feature(enable = "neon")]
+    pub unsafe fn dot(u: ArrayView1<f32>, v: ArrayView1<f32>) -> f32 {
+        assert_eq!(u.len(), v.len());
+
+        let mut u = u
+            .as_slice()
+            .expect("Cannot apply SIMD instructions on non-contiguous data.");
+        let mut v = &v
+            .as_slice()
+            .expect("Cannot apply SIMD instructions on non-contiguous data.")[..u.len()];
```

Comment:
`u.len() == v.len()` is asserted above, is there a reason to slice the slice again here?

Reply:
I copied over the non-SIMD parts from one of the other functions. At some point in time, I had to add the additional slicing, because the `assert` alone would not convince the compiler to eliminate bounds checking. Maybe this has changed, I'll have to check.

Reply:
That's a fair reason, just looked like some accidental code remnants!